erpnext_shipping/erpnext_shipping/shipping.py

```diff
@@ -195,36 +364,56 @@ def print_shipping_label(shipment: str):
 		_labels = sendcloud.get_label(shipment_id)
 		for i, label_url in enumerate(_labels, start=1):
 			content = sendcloud.download_label(label_url)
-			file_url = save_label_as_attachment(shipment, content, i)
+			file_url = save_label_as_attachment(shipment=shipment, content=content, index=i)
 			shipping_label.append(file_url)
+	elif service_provider == SHIPROCKET_PROVIDER:
+		shipping_label = []
+		shiprocket = ShiprocketUtils(company=pickup_company)
+		label_url = shiprocket.generate_label(shipment_id, shipment)
+		if not label_url:
+			frappe.throw(_("Failed to generate label."))
+		file_url = save_label_as_attachment(shipment=shipment, url=label_url)
+		shipping_label.append(file_url)
+	elif service_provider == ENVIA_PROVIDER:
+		shipping_label = []
+		file_url = get_shipping_label(shipment)
+		if not file_url:
+			frappe.throw(_("Failed to generate label."))
+		shipping_label.append(file_url)
+	elif service_provider == DELHIVERY_PROVIDER:
+		delhivery = DelhiveryOneUtils(company=pickup_company)
+		shipping_label = delhivery.get_label(shipment_id)
+	elif service_provider == SHIPPO_PROVIDER:
+		shipping_label = []
+		shippo = ShippoUtils(company=pickup_company)
+		label_url = shippo.get_label(shipment_id, shipment)
+		if not label_url:
+			frappe.throw(_("Failed to generate label."))
+		file_url = save_label_as_attachment(shipment=shipment, url=label_url)
+		shipping_label.append(file_url)
+	elif service_provider == ARAMEX_PROVIDER:
+		shipping_label = []
+		file_url = get_shipping_label(shipment)
+		if not file_url:
+			frappe.throw(_("Failed to generate label."))
+		shipping_label.append(file_url)
```

Comment:
I'd expect that all of these to look and behave roughly the same. Right now, some raise an error and some don't; some save as attachment and some don't.

Also, could we pull out the "Failed to generate label" check to the end, just before the return statement?